File: frontend/src/types/assessment.ts
export type AssessmentType = 'phq9' | 'gad7' | 'stress';
export type SeverityLevel = 'low' | 'moderate' | 'high';

export interface QuestionOption {
  value: number;
  label: string;
}

export interface Question {
  id: string;
  text: string;
}

export interface AssessmentQuestions {
  assessment_type: AssessmentType;
  title: string;
  description: string;
  instructions: string;
  options: QuestionOption[];
  questions: Question[];
  max_score: number;
}

export interface AssessmentResult {
  id: string;
  assessment_type: AssessmentType;
  title: string;
  score: number;
  max_score: number;
  severity: SeverityLevel;
  severity_label: string;
  wellness_tips: string[];
  completed_at: string;
}

export interface AssessmentHistoryItem {
  id: string;
  assessment_type: AssessmentType;
  title: string;
  score: number;
  max_score: number;
  severity: SeverityLevel;
  completed_at: string;
}

export interface AssessmentSummary {
  latest: Record<AssessmentType, AssessmentResult | null>;
  history_count: number;
}

export interface TrendPoint {
  date: string;
  score: number;
  severity: SeverityLevel;
}

export interface AssessmentTrends {
  phq9: TrendPoint[];
  gad7: TrendPoint[];
  stress: TrendPoint[];
}

export const ASSESSMENT_META: Record<
  AssessmentType,
  { title: string; description: string; color: string }
> = {
  phq9: {
    title: 'Depression Screening',
    description: 'PHQ-9 based questionnaire to screen for depressive symptoms',
    color: 'calm',
  },
  gad7: {
    title: 'Anxiety Screening',
    description: 'GAD-7 based questionnaire for anxiety symptoms',
    color: 'green',
  },
  stress: {
    title: 'Perceived Stress (PSS-10)',
    description: '10-item Perceived Stress Scale for stress over the past month',
    color: 'calm',
  },
};

File: backend/app/schemas/assessment.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import AssessmentType


class QuestionOption(BaseModel):
    value: int
    label: str


class Question(BaseModel):
    id: str
    text: str


class AssessmentQuestionsResponse(BaseModel):
    assessment_type: str
    title: str
    description: str
    instructions: str
    options: list[QuestionOption]
    questions: list[Question]
    max_score: int


class SubmitAssessmentRequest(BaseModel):
    assessment_type: AssessmentType
    responses: dict[str, int] = Field(..., description="Map of question id to score value")


class AssessmentResultResponse(BaseModel):
    id: str
    assessment_type: str
    title: str
    score: int
    max_score: int
    severity: str
    severity_label: str
    wellness_tips: list[str]
    completed_at: str


class AssessmentHistoryItem(BaseModel):
    id: str
    assessment_type: str
    title: str
    score: int
    max_score: int
    severity: str
    completed_at: str


class AssessmentSummaryResponse(BaseModel):
    latest: dict[str, AssessmentResultResponse | None]
    history_count: int


class TrendPoint(BaseModel):
    date: str
    score: int
    severity: str


class AssessmentTrendsResponse(BaseModel):
    phq9: list[TrendPoint]
    gad7: list[TrendPoint]
    stress: list[TrendPoint]
